inc/shortcodes/class-pdf.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Shortcake_Bakery\Shortcodes;
+
+class PDF extends Shortcode {
+
+	public static function get_shortcode_ui_args() {
+		return array(
+			'label'          => esc_html__( 'PDF', 'shortcake-bakery' ),
+			'listItemImage'  => 'dashicons-media-document',
+			'attrs'          => array(
+				array(
+					'label'  => esc_html__( 'URL', 'shortcake-bakery' ),
+					'attr'   => 'url',
+					'type'   => 'text',
+				),
+			),
+		);
+	}
+
+	/**
+	 * Turn embed code into a proper shortcode
+	 *
+	 * @param  string $content
+	 * @return string $content
+	 */
+	public static function reversal( $content ) {
+		$needle = '#http[^<]//[^<]+\.pdf?#';
```

Comment:
Checking for the string `pdf` is a good suggestion.

That regex looks like it will match all kinds of things that aren't a linkable pdf, like if I wrote:

> I'm thinking about http,// you know something something .pdf or other..

It also won't match https urls. But other than fixing the regex, it looks good to me...